homeassistant/components/websocket_api.py

```diff
@@ -348,43 +348,51 @@ def cancel_connection(event):
 
         return wsock
 
+    @asyncio.coroutine
     def handle_subscribe_events(self, msg):
         """Handle subscribe events command."""
         msg = SUBSCRIBE_EVENTS_MESSAGE_SCHEMA(msg)
 
-        @callback
+        @asyncio.coroutine
         def forward_events(event):
             """Forward events to websocket."""
             if event.event_type == EVENT_TIME_CHANGED:
                 return
 
             try:
-                self.send_message(event_message(msg['id'], event))
+                yield from self.send_message(event_message(msg['id'], event))
             except RuntimeError:
                 # Socket has been closed.
                 pass
 
         self.event_listeners[msg['id']] = self.hass.bus.async_listen(
             msg['event_type'], forward_events)
 
-        self.send_message(result_message(msg['id']))
+        return self.send_message(result_message(msg['id']))
```

Comment:
All handle_* methods are yielded from. So they either have to be a coroutine or return one. Trying to save yields.